Various updates

DavidRieman opened the pull request:
Default dictionary updated, hotkey command addressed, katu07's localization support, and cleanup. 
Works on Win 10 but the taskbar likes to show.

shanselman commented:
Cool! This will take me a while to look through...

Razzeee commented:
Any news on this, I would be especially interested in the localization feature and willing to add german.

shanselman commented:
Eek! I need some help, I'm travelling. Someone want to be a merger/viewer? @Razzeee @ericlaw1979

Razzeee commented:
Well maybe we can get @DavidRieman to split these additions into more PRs? It's a pretty big PR with alot of commits/code changed.

DavidRieman commented:
The GitHub "Commits" tab produces a very nice way to review one commit at a time, since my commit workflow is pretty granular in that way.  I would be happy to address code review comments made at a per-commit level.  (I would prefer to avoid the overhead of making a bunch of new branches and cherry-picking commits to push and make new pull requests, if reviewing per commit is a suitable workflow, so let me know.)

Razzeee commented:
I understand. I'm aware of the github feature here I just know and like the practice to separate prs by topic. As that might lead to the trivial stuff getting merged faster and not getting blocked by stuff that needs discussion.

shanselman commented:
Do you just want commit, @DavidRieman or should I just merge this?

drewnoakes commented:
Not sure 33812e5 is a good idea generally. You can attach binaries to 'releases' on GitHub without adding them to the repo.

drewnoakes commented on BabySmash.csproj:
```diff
@@ -10,14 +10,14 @@
     <RootNamespace>BabySmash</RootNamespace>
     <AssemblyName>BabySmash</AssemblyName>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
What's the reason to increase the minimum supported framework? The higher this number is, the fewer people can run this.

DavidRieman replied:
Not sure if it was intended.  Those were merged in from @katu07 language support.  I've pulled it back to 3.5 now.

DavidRieman commented:
I believe I have addressed all the PR feedback.  Let me know if I missed something.
* Zip is gone.
* Target .NET reverted to 3.5.
* SignManifests is reverted.  I disagree with this, but I will open a new issue to discuss; the issue need not bog down this PR  :)

DavidRieman commented:
@shanselman I'm not sure what your request was trying to communicate.  If it is "would you like to address PR feedback, or should I just merge?" then the former.

drewnoakes commented:
@DavidRieman it'd be better to squash/fixup the add/remove commits so the zip doesn't end up in the repo history. You can do this with an interactive rebase, then force push.

shanselman commented:
So should I merge now or wait @drewnoakes ?

drewnoakes commented:
Up to you. If you accept the PR by squashing all commits (via the dropdown) then the binaries won't end up in the history. Instead you'll end up with one large commit that loses the granular messages @DavidRieman put against each commit. If it were my project I'd wait for the two commits to be squashed (or do it myself). I didn't see how large the binaries are, but right now your repo is nice and small.

DavidRieman commented:
I'd be happy to look into squashing on my end;  I'm long overdue for learning how to do that anyway.  It may be another day or so before I can dive into that though.

drewnoakes commented:
Look into interactive rebase. For example:

    git rebase -i shanselman/master

This launches an editor with a script that you can edit to modify how commits are rebased. You can reorder, drop commits, squash, and so forth. It's very powerful. Save and exit the editor to start the process. You might hit conflicts as it plays the script, which you'd have to resolve.

Once done you can push your branch. Because you'll have edited the history, you'll need to force push.

Hope that helps.

DavidRieman commented:
Ok, rebase is still a little scary to me but I got the job done with:
  git filter-branch --tree-filter 'rm -f BabySmashInternational.zip' HEAD
which nuked all trace of the file from those commits without messing with the existence of the commits themselves.

shanselman commented:
So do I merge now?

DavidRieman commented:
My end is All Clear for merge.
Sign off from @Razzeee or @drewnoakes would be the proper next step.
